### PR TITLE
test union

### PR DESCRIPTION
**Pull request by RemHero**

### CODE REVIEW

**Review by YouJiacheng — APPROVED** [review]

**Review by Cauchy-NY — APPROVED** [review]

**Review by zhuyuqing — CHANGES_REQUESTED** [review]
Please fix all comments sent to you. Thanks.

File: .github/workflows/IoTDB-CE.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      #第 3 阶段测试开始==========================================
+      - name: Prepare CapExp environment
+        if: always()
+        uses: ./.github/actions/capacityExpansionUnionTest
+        with:
+          version: ${VERSION}
+          DB-name: ${{matrix.DB-name}}
+          Test-Way: oriHasDataExpNoData
+      - name: oriHasDataExpNoData IT
+        if: always()
```
Comment (YouJiacheng) [inline .github/workflows/IoTDB-CE.yml:89]: Can we use check `matrix.DB-name` here?

File: .github/workflows/IoTDB-CE.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        run: |
+          mv test/pom.xml test/pom.xml.backup
+          mv test/pom.xml.${{matrix.DB-name}} test/pom.xml
+      - name: Install IginX with Maven
+        shell: bash
+        run: |
+          mvn clean package -DskipTests
+      #第 1 阶段测试开始==========================================
+      - name: Prepare CapExp environment
+        if: always()
```
Comment (YouJiacheng) [inline .github/workflows/IoTDB-CE.yml:48]: Can we check `matrix.DB-name` here?

File: .github/workflows/IoTDB-CE.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        run: |
+          if [ "${{matrix.DB-name}}" == "iotdb12" ]; then
+            mvn test -q -Dtest=IoTDB12HistoryDataCapacityExpansionIT#oriHasDataExpHasData -DfailIfNoTests=false
+            mvn test -q -Dtest=IoTDBTagScaleOutIT,IoTDBRestfulScaleOutIT,IoTDBSessionScaleOutIT,IoTDBSqlScaleOutIT -DfailIfNoTests=false
+          else
+            echo "${{matrix.DB-name}} is not supported"
+          fi
+      #第 2 阶段测试开始==========================================
+      - name: Prepare CapExp environment
```
Comment (YouJiacheng) [inline .github/workflows/IoTDB-CE.yml:64]: Duplicated name?

Comment (YouJiacheng) [reply .github/workflows/IoTDB-CE.yml:64]: Maybe "Prepare CapExp environment for oriNoDataExpNoData"

File: .github/workflows/IoTDB-CE.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      - name: oriHasDataExpHasData IT
+        if: always()
+        run: |
+          if [ "${{matrix.DB-name}}" == "iotdb12" ]; then
+            mvn test -q -Dtest=IoTDB12HistoryDataCapacityExpansionIT#oriHasDataExpHasData -DfailIfNoTests=false
+            mvn test -q -Dtest=IoTDBTagScaleOutIT,IoTDBRestfulScaleOutIT,IoTDBSessionScaleOutIT,IoTDBSqlScaleOutIT -DfailIfNoTests=false
+          else
+            echo "${{matrix.DB-name}} is not supported"
+          fi
+      #第 2 阶段测试开始==========================================
```
Comment (YouJiacheng) [inline .github/workflows/IoTDB-CE.yml:63]: Does "oriNoDataExpNoData IT" depend on "oriHasDataExpHasData IT"? Maybe we can run them in parallel to reduce feedback time.

**Pull request closed without merging**